vcd/datasource_vcd_dse_registry_configuration.go

```diff
@@ -0,0 +1,121 @@
+package vcd
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func datasourceVcdDseRegistryConfiguration() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: datasourceVcdDseRegistryConfigurationRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Data Solution Name",
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Type of Data Solution package. One of 'PackageRepository', 'ChartRepository'",
+			},
+			"package_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Package name",
+			},
+			"default_package_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Package name when type=ChartRepository",
+			},
+			"chart_repository": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Chart repository used",
+			},
+			"default_chart_repository": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Default chart repository provided by Data Solution",
+			},
+			"version": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Version of package to use",
+			},
+			"default_version": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Default version provided by Data Solution",
+			},
+			"package_repository": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Package repository to use",
+			},
+			"default_repository": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Default package repository provided by Data Solution",
+			},
+			"container_registry": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Description: "Container registry configuration - only applicable for 'VCD Data Solutions'",
+				Elem:        datasourceDseContainerRegistry,
+			},
+			"compatible_version_constraints": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "A set of version compatibility constraints",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"requires_version_compatibility": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Boolean flag if the Data Solution requires version compatibility",
+			},
+			"rde_state": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Parent RDE state",
+			},
+		},
+	}
+}
+
+var datasourceDseContainerRegistry = &schema.Resource{
```

Comment:
Are these Required/Optional, or just Computed?

Reply:
This is a schema for data source - so yes - all computed.